### ISSUE TITLE
Improve Mouse wheel usage for plugin cards

### ISSUE THREAD

**Issue by pStyl3** [loot/loot#1679]
Would it be possible so that the plugin card list can be moved by pressing down the mouse wheel? Pressing down the mouse wheel currently doesn't have an effect in the plugin card list.

Also, I think that the normal scrolling using the mouse wheel scrolls LOOTs right pane too much up and down at once. Here's a screenshot with the general information card being at the top, notice the placement of `ccbgssse005-goldbrand.esl`:
![grafik](https://user-images.githubusercontent.com/10406657/150838087-65b68522-ee90-4588-8dd3-2d939704363d.png)

And here is one after I scrolled down once using the mouse wheel:
![grafik](https://user-images.githubusercontent.com/10406657/150838246-608f27b4-539f-40e3-aacc-0c0f5e6e0d79.png)

`ccbgssse005-goldbrand.esl` is now at the top, almost out of view. Would it be possible to decrease the amount those incremental steps are moving the list?

**Comment by Ortham** [comment]
I've tweaked the scrolling step size in the `better-card-scrolling` branch, what do you think of it now?

**Comment by pStyl3** [comment]
Using `loot_0.17.0-97-g78a3935_better-card-scrolling-win64.7z`, I think scrolling the plugin cards is now a lot better than before, I'm now moving the list by ~3 plugins with each scroll. Personally I probably would reduce it even more, just a bit, so that the list get's moved by ~2 plugin cards. The Qt GUI is really compact now, so the increased need for scrolling shouldn't matter too much, however you should gain more control in moving the plugin cards exactly where you want them to be, and it should be getting easier to "follow the plugin cards with your eyes".

**Comment by sibir-ine** [comment]
I like the new scrolling speed as well 👍

**Comment by Ortham** [comment]
@pStyl3 @sibir-ine Are you sure the scroll speed actually changed? The commit you downloaded the snapshot build for has a bug that means there should be (and as far as I can tell, there isn't) no change in scrolling behaviour.

**Comment by pStyl3** [comment]
Yes, the speed did indeed change. I've recorded two videos to show the difference (the following links will be deprecated once this issue is closed).

`loot_0.17.0-84-g864b91b_master-win64.7z` (from [here](https://github.com/loot/loot/actions/runs/1714905280)):
[Video 1](https://mega.nz/file/WxZjRYhZ#6-NIz7ggh4TTuxcdMnUxNc5KT-m3vzM1WUnmgbQpwI4)

`loot_0.17.0-97-g78a3935_better-card-scrolling-win64.7z`:
[Video 2](https://mega.nz/file/31Jx2AhT#6fODWx-reWLEhKHua3vz8lhDNLDh6fuiH9QwZYM9Xz4)

I've had the About LOOT window and the search window open during the recording, but they apparently didn't get captured, only LOOTs main window did. Video 1 is the reference, Video 2 shows how scrolling is happening now in smaller steps.

**Comment by sibir-ine** [comment]
Yeah, the speed has in fact changed for me too. Before, in v0.17.0+735ab41, one scroll would move the card list an entire page, including messages. Now, in v0.17.0+78a3935, one scroll moves 3 cards at a time.

**Comment by Ortham** [comment]
Huh, I tested again and I see the same thing.

**Comment by Ortham** [comment]
> Would it be possible so that the plugin card list can be moved by pressing down the mouse wheel? Pressing down the mouse wheel currently doesn't have an effect in the plugin card list.

By this, I think you mean auto-scroll (pressing and holding the middle mouse button shows an icon with some arrows, and then moving the mouse in a direction scrolls in that direction)? There doesn't seem to be any built-in support for it, and while I might be able to implement it on a case-by-case basis, I think that would just end up with some things not being scrollable like that, and the inconsistency might be worse than it just not being supported at all.

**Comment by pStyl3** [comment]
> By this, I think you mean auto-scroll (pressing and holding the middle mouse button shows an icon with some arrows, and then moving the mouse in a direction scrolls in that direction)?

That's correct. It would be a nice to have feature, though I can also understand your reasoning, if it's too much of a hassle to implement.

**Comment by Ortham** [comment]
I've just updated the `better-card-scrolling` branch to scroll the cards by a fixed amount that's the height of two empty cards, what do you think of it now?

**Comment by pStyl3** [comment]
That's much better, thanks. 👍

**Comment by Ortham** [comment]
OK, I'll close this then as I've merged that change in 695795782bf62e9890338ffd3c6ae17660a7bdfa, and I'm not going to look into auto-scroll any further than I have.

**Comment by pStyl3** [comment]
I've noticed that there's one more difference between the old and new GUI when it comes to scrolling the plugin cards. The current Qt GUI scrolls the intended amount _instantly_, while with the old GUI scrolling once would take roughly about 1/4th of a second and the scrolling itself would _transition smoothly_ from the current to the new position.

Now that I've noticed this, I think that this transition effect seems to really help in following a point (plugin card) with your eyes moving up and down in the list, since this visual effect creates a real motion which the eye can follow.

I think this would be great to have.

**Comment by Ortham** [comment]
I don't think that would be easy to implement, but I'll reopen this so that suggestion isn't lost.

**Comment by pStyl3** [comment]
For what it's worth, multiple programs I know also made their scrolling behavior smooth & a bit delayed, e.g. Discord, Word, Firefox. Now that doesn't really help here since AFAIK those apps aren't based upon Qt, but still it doesn't seem to be a rare feature.

Seaching `Qt smooth scrolling` via google yields a few results, e.g. [this](https://stackoverflow.com/questions/2016323/qt4-is-it-possible-to-make-a-qlistview-scroll-smoothly/2016538) one (though I don't know how much that would help).

**Comment by Ortham** [comment]
I do have an idea of how to implement smooth scrolling (basically, intercept the scroll event, measure how far it would scroll, cancel the event and then scroll that distance in smaller steps over a period of time). As I see it the complexity is more around dealing with things like a user that keeps scrolling (you've got to add the extra scrolling distances to the smooth scroll that is currently happening) and more significantly how to tune the smooth scrolling so that it feels good.
